app/controllers/FileChecksController.scala

```diff
@@ -37,39 +43,70 @@ class FileChecksController @Inject() (
       }
   }
 
+  private def handleFailedUpload(consignmentId: UUID, consignmentRef: String, isJudgmentUser: Boolean, token: BearerAccessToken)(implicit request: Request[AnyContent]) = {
+    for {
+      _ <- consignmentStatusService.updateConsignmentStatus(ConsignmentStatusInput(consignmentId, UploadType.id, Some(CompletedWithIssuesValue.value)), token)
+    } yield {
+      Ok(views.html.uploadInProgress(consignmentId, consignmentRef, "Uploading your records", request.token.name, isJudgmentUser))
+        .uncache()
+    }
+  }
+
+  private def triggerBackendChecks(consignmentId: UUID, token: BearerAccessToken): Future[Boolean] = {
+    for {
+      backendChecksTriggered <- backendChecksService.triggerBackendChecks(consignmentId, token.getValue)
+      uploadStatusUpdate =
+        if (backendChecksTriggered) {
+          CompletedValue
+        } else {
+          CompletedWithIssuesValue
+        }
+      _ <- consignmentStatusService.updateConsignmentStatus(ConsignmentStatusInput(consignmentId, UploadType.id, Some(uploadStatusUpdate.value)), token)
+    } yield backendChecksTriggered
+  }
+
   def fileCheckProgress(consignmentId: UUID): Action[AnyContent] = secureAction.async { implicit request =>
     consignmentService
       .fileCheckProgress(consignmentId, request.token.bearerAccessToken)
       .map(_.asJson.noSpaces)
       .map(Ok(_))
   }
 
-  def fileChecksPage(consignmentId: UUID): Action[AnyContent] = standardTypeAction(consignmentId) { implicit request: Request[AnyContent] =>
+  def fileChecksPage(consignmentId: UUID, uploadFailed: Option[String]): Action[AnyContent] = standardTypeAction(consignmentId) { implicit request: Request[AnyContent] =>
+    val token = request.token.bearerAccessToken
     for {
-      fileChecks <- getFileChecksProgress(request, consignmentId)
-      reference <- consignmentService.getConsignmentRef(consignmentId, request.token.bearerAccessToken)
-    } yield {
-      if (fileChecks.isComplete) {
-        Ok(
-          views.html.fileChecksProgressAlreadyConfirmed(
-            consignmentId,
-            reference,
-            applicationConfig.frontEndInfo,
-            request.token.name,
-            isJudgmentUser = false
-          )
-        ).uncache()
-      } else {
-        Ok(views.html.standard.fileChecksProgress(consignmentId, reference, applicationConfig.frontEndInfo, request.token.name))
-          .uncache()
+      reference <- consignmentService.getConsignmentRef(consignmentId, token)
+      result <- uploadFailed match {
+        case Some(TRUE) => handleFailedUpload(consignmentId, reference, isJudgmentUser = false, token)
```

Comment:
Don't use the apache `TRUE`. Just use `case Some("true") ...` Or write a common utils method to convert a string option to boolean.